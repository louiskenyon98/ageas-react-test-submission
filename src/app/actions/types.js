//Define all types in app as constants, and export.

export const GET_NOTES = 'GET_NOTES';
export const GET_NOTE = 'GET_NOTE';
export const POST_NOTE = 'POST_NOTE';
export const PATCH_NOTE = 'PATCH_NOTE';
export const DELETE_NOTE = 'DELETE_NOTE';
export const SHOW_HELLO_WORLD_MODAL = 'SHOW_HELLO_WORLD_MODAL';
export const CLOSE_MODAL = 'RESET_MODAL';

