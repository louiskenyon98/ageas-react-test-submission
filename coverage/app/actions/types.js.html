<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/actions/types.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">app/actions</a> types.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">//Define all types in app as constants, and export.
&nbsp;
export const GET_NOTES = 'GET_NOTES';
export const GET_NOTE = 'GET_NOTE';
export const SHOW_STATUS_MODAL = 'SHOW_STATUS_MODAL';
export const SHOW_CONFIRMATION_MODAL = 'SHOW_CONFIRMATION_MODAL';
export const ONCHANGE_FILTER_VALUE = 'ONCHANGE_FILTER_VALUE';
export const CLOSE_MODAL = 'RESET_MODAL';
&nbsp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 18 2019 10:21:20 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
