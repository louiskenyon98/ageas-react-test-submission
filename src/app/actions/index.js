import axios from 'axios';
import history from '../history';
import {GET_NOTE, GET_NOTES, POST_NOTE, PATCH_NOTE, DELETE_NOTE, API_ERROR} from "./types";

//I have only added error handling to this route, but I have added it to App.jsx.
//Thereby assuring that I can only use the application if the API is functioning.
//Action creator to get all notes.
export const getNotes = () => async dispatch => {
    try {
        const response = await axios.get('/api/notes/');
        dispatch({
            type: 'GET_NOTES',
            payload: response.data
        });
    } catch (error) {
        dispatch({
            type: 'API_ERROR'
        })
    }
};

//Couldn't figure out which way was the best way to write this, so I wrote it both ways for posterity.

// export const getNotes = () => async dispatch => {
//     const response = await axios.get('/api/notes/')
//         .then(function(response) {
//             dispatch({
//                 type: 'GET_NOTES',
//                 payload: response.data
//             });
//         })
//         .catch(function() {
//             dispatch({
//                 type: 'API_ERROR'
//             })
//         })
// };

//Action creator to get a single note.
export const getNote = (id) => async dispatch => {
    const response = await axios.get(`/api/notes/${id}`);
    dispatch({
        type: 'GET_NOTE',
        payload: response.data
    });
};
//Action creator to post a new note
export const postNote = (data) => async dispatch => {
    const response = await notesAPI.post('/api/notes/', data);
    dispatch({
        type: 'POST_NOTE',
        payload: response.data
    });
    //Programmatic navigation of user back to AllNotes after form submission is completed.
    history.push('/');
};
export const patchNote = (data) => async dispatch => {
    const response = await notesAPI.patch('/api/notes/', data);
    dispatch({
        type: 'PATCH_NOTE',
        payload: response.data
    });
    //See comment in previous action creator.
    history.push('/');
};

export const deleteNote = (id) => async dispatch => {
    const response = await notesAPI.delete(`/api/notes/${id}`, null);
    dispatch({
        type: 'DELETE_NOTE',
        payload: response.data.id
    });
};
