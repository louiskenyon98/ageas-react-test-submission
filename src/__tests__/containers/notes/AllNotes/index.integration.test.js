import React from 'react';
import {mount} from 'enzyme';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import {Router} from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
import AllNotesContainer from '../../../../app/containers/notes/AllNotes/index';
import {getNotes} from '../../../../app/actions/note/notes.action.js'
import {SHOW_DELETE_CONFIRMATION_MODAL} from "../../../../app/actions/types";
// import {showDeleteNoteConfirmationModal} from '../../../../app/actions/common/modal.action';

jest.mock('../../../../app/actions/note/notes.action', () => ({
    getNotes: jest.fn(() => () => {
    })
}));

describe('AllNotesContainer Integration Test', () => {
    let props = {};
    let store = {};
    let mountedComponent;
    let history = createHistory();
    const middleware = [thunk];
    const mockStore = configureMockStore(middleware);
    let wrapper = () => {
        if (!mountedComponent) {
            mountedComponent = mount(
                <Provider store={store}>
                    <Router history={history}>
                        <AllNotesContainer {...props}/>
                    </Router>
                </Provider>
            )
        }
        return mountedComponent
    };
    beforeEach(() => {
        props = {};
        store = mockStore({
            notes: {
                data: [
                    {
                        id: 90,
                        title: "title number 1",
                        body: "the first body"
                    },
                    {
                        id: 91,
                        title: "title number 2",
                        body: "the second body"
                    }
                ]
            }
        });
        mountedComponent = undefined;
        jest.clearAllMocks();
    });
    describe('render', () => {
        it('should render AllNotesContainer', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        it('should call getNotes on mount', () => {
            expect(getNotes).toHaveBeenCalledTimes(0);
            wrapper();
            expect(getNotes).toHaveBeenCalledTimes(1);
        });
        it('should call showDeleteConfirmationModal on confirmDelete click', () => {
            expect(store.getActions().length).toBe(0);
            wrapper().find('AllNotes').find('Card').find('.content').find('i').first().simulate('click');
            expect(store.getActions().length).toBe(1);
            expect(store.getActions()[0]).toEqual({
                type: 'SHOW_DELETE_CONFIRMATION_MODAL',
                payload: {
                    id: 90
                }
            })
        })
    })
});
