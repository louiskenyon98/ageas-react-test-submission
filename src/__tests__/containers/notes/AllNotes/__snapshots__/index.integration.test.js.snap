// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllNotesContainer Integration Test render should render AllNotesContainer 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(AllNotesContainer)>
      <AllNotesContainer
        getNotes={[Function]}
        notes={
          Array [
            Object {
              "body": "the first body",
              "id": 90,
              "title": "title number 1",
            },
            Object {
              "body": "the second body",
              "id": 91,
              "title": "title number 2",
            },
          ]
        }
        showDeleteNoteConfirmationModal={[Function]}
      >
        <AllNotes
          confirmDelete={[Function]}
          data={
            Array [
              Object {
                "body": "the first body",
                "id": 90,
                "title": "title number 1",
              },
              Object {
                "body": "the second body",
                "id": 91,
                "title": "title number 2",
              },
            ]
          }
        >
          <div
            className="container-fluid"
          >
            <div
              className="row"
            >
              <Card
                body="the first body"
                delete={[Function]}
                id={90}
                key="90"
                title="title number 1"
              >
                <div
                  className="col-xs-1 col-md-4 col-xl-3 d-flex align-items-stretch column"
                >
                  <div
                    className="ui raised card card"
                  >
                    <div
                      className="content"
                    >
                      <i
                        className="right floated close icon delete-button"
                        onClick={[Function]}
                      />
                      <div
                        className="header header"
                      >
                        title number 1
                      </div>
                      <div
                        className="description body"
                      >
                        <p>
                          the first body
                        </p>
                      </div>
                    </div>
                    <div
                      className="extra content"
                    >
                      <span
                        className="right floated pencil alternate"
                      >
                        <Link
                          replace={false}
                          to="/edit/90"
                        >
                          <a
                            href="/edit/90"
                            onClick={[Function]}
                          >
                            <i
                              className="pencil alternate icon edit-button"
                            />
                          </a>
                        </Link>
                      </span>
                    </div>
                  </div>
                </div>
              </Card>
              <Card
                body="the second body"
                delete={[Function]}
                id={91}
                key="91"
                title="title number 2"
              >
                <div
                  className="col-xs-1 col-md-4 col-xl-3 d-flex align-items-stretch column"
                >
                  <div
                    className="ui raised card card"
                  >
                    <div
                      className="content"
                    >
                      <i
                        className="right floated close icon delete-button"
                        onClick={[Function]}
                      />
                      <div
                        className="header header"
                      >
                        title number 2
                      </div>
                      <div
                        className="description body"
                      >
                        <p>
                          the second body
                        </p>
                      </div>
                    </div>
                    <div
                      className="extra content"
                    >
                      <span
                        className="right floated pencil alternate"
                      >
                        <Link
                          replace={false}
                          to="/edit/91"
                        >
                          <a
                            href="/edit/91"
                            onClick={[Function]}
                          >
                            <i
                              className="pencil alternate icon edit-button"
                            />
                          </a>
                        </Link>
                      </span>
                    </div>
                  </div>
                </div>
              </Card>
            </div>
          </div>
        </AllNotes>
      </AllNotesContainer>
    </Connect(AllNotesContainer)>
  </Router>
</Provider>
`;
